### PR TITLE
LCOW: Fix nits from 33241

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard <jhoward@microsoft.com>

Fixes most of the feedback 'nits' comments from @johnstep in #33241

### CODE REVIEW

**Review by johnstep — APPROVED** [review]
LGTM

File: api/server/backend/build/tag.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func (bt *Tagger) TagImages(imageID image.ID) error {
 	for _, rt := range bt.repoAndTags {
 		// TODO @jhowardmsft LCOW support. Will need revisiting.
 		platform := runtime.GOOS
-		if platform == "windows" && system.LCOWSupported() {
+		if system.LCOWSupported() {
```
Comment (johnstep) [inline api/server/backend/build/tag.go]: What about the other instances of this redundant (`platform == "windows"`) check?

File: builder/dockerfile/parser/parser.go [hunk 2]
```diff
@@ -177,7 +177,7 @@ func (d *Directive) possibleParserDirective(line string) error {
 func NewDefaultDirective() *Directive {
 	directive := Directive{}
 	directive.setEscapeToken(string(DefaultEscapeToken))
-	directive.setPlatformToken(runtime.GOOS)
+	directive.setPlatformToken(DefaultPlatformToken)
```
Comment (thaJeztah) [inline builder/dockerfile/parser/parser.go]: I noticed `DefaultPlatformToken` is only used in this file; can you un-export it?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
left one nit, but looks good otherwise

File: builder/dockerfile/parser/parser.go (hunk 2 above)
Comment (jhowardmsft) [inline builder/dockerfile/parser/parser.go]: Yup, fixed. Push coming shortly.

File: api/server/backend/build/tag.go (hunk 1 above)
Comment (jhowardmsft) [inline api/server/backend/build/tag.go]: As mentioned on slack, it's a little moot as I'm actively working to remove all these markers anyway. But done. Push imminent.

**Comment by jhowardmsft** [comment]
Comments addressed.

**Review by thaJeztah — APPROVED** [review]
LGTM

ping @johnstep PTAL

**Review by johnstep — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 👼

**Comment by cpuguy83** [comment]
Failures look legit.

**Comment by jhowardmsft** [comment]
@cpuguy83 yup, they do. Investigating

**Comment by jhowardmsft** [comment]
I see it. Should be fixed in latest push.

**Comment by jhowardmsft** [comment]
Experimental failure (just locked up and about to timeout) is unrelated. Restarting for the 3rd time.

**Comment by jhowardmsft** [comment]
Green 💚 . Finally. (All sorts of CI infrastructure issues on multiple contexts the past few days)

**Pull request merged**